src/app/icd/ICDMonitoringTable.h

```diff
@@ -59,6 +59,25 @@ struct ICDMonitoringEntry : public PersistentData<kICDMonitoringBufferSize>
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    /**
+     * @brief Set the Key object
+     *        This method will create a new keyHandle. The key handle might contain either
+     *        the raw key or a keyID depending on which Crypto implementation is used.
+     *        In any case, to prevent key leakage, one should either call the DeleteKey method
+     *        or save the entry within the ICDMonitoring Table before this object goes out of scope.
```

Comment:
```suggestion
     *        To avoid leaking keys, API consumers must either call the DeleteKey method
     *        or save the entry within the ICDMonitoring Table before this object goes out of scope.
```

Reply:
fixed